### ISSUE TITLE
Softmax layer for output probabilities 

### ISSUE THREAD

**Issue by RobinVdE** [jadore801120/attention-is-all-you-need-pytorch#58]
Hello,

In the main Transformer model the encoder and decoder parts are calculated.
Then they are fed to a linear layer for the target word projections. But shouldn't this layer be followed by a softmax function to calculate the output probabilities like in the Transformer schematic?

Or am I looking over something? I can't seem to locate this last softmax function in the code.

**Comment by yalunar** [comment]
Hello, @RobinVdE. I think it's because in the [train.py](https://github.com/jadore801120/attention-is-all-you-need-pytorch/blob/7fa8c63f9a5a6e997659054ebef2fc1a61630dc2/train.py#L262) , the loss function need the original outputs. You can see the [cross_entropy](https://pytorch.org/docs/stable/nn.html?highlight=cross_entropy#torch.nn.functional.cross_entropy) in this document, it combines `log_softmax` and `nll_loss` , in which the `log_softmax` function has implemented the softmax function.

**Comment by jadore801120** [comment]
Hello @yalunar, thanks for the explanation!
Hello @RobinVdE, if you have other questions, feel free to reopen the issue.
Thanks!
Yu-Hsiang
